src/Akeneo/Connectivity/Connection/back/Infrastructure/EventSubscriber/ReadProductsEventSubscriber.php

```diff
@@ -47,24 +49,56 @@ public static function getSubscribedEvents(): array
      */
     public function saveReadProducts(ReadProductsEvent $event): void
     {
-        if (!$this->connectionContext->isCollectable()) {
-            return;
-        }
-        if (0 === count($event->productIds())) {
+        if (0 === $event->getCount()) {
             return;
         }
+        $connectionCode = $this->getConnectionCodeByReadProductsEvent($event);
 
-        $connection = $this->connectionContext->getConnection();
-        if (FlowType::DATA_DESTINATION !== (string) $connection->flowType()) {
-            return;
+        if (!$this->connectionIsValid($connectionCode, $event->isEventApi())) {
+            throw new \LogicException('The connection is not valid.');
         }
 
         $this->updateDataDestinationProductEventCountHandler->handle(
             new UpdateDataDestinationProductEventCountCommand(
-                (string) $connection->code(),
+                $connectionCode,
                 HourlyInterval::createFromDateTime(new \DateTimeImmutable('now', new \DateTimeZone('UTC'))),
-                count($event->productIds())
+                $event->getCount()
             )
         );
     }
+
+    private function getConnectionCodeByReadProductsEvent(ReadProductsEvent $event): ?string
+    {
+        $connection = $this->connectionContext->getConnection();
+
+        return $event->isEventApi()
+            ? $event->getConnectionCode()
+            : ($connection === null ? null : $connection->code()->__toString());
```

Comment:
By fetching the connection first, you execute 2 sql queries, even in the case you don't need it !
A more performant approach would be something like this:
```suggestion
        if ($event->isEventApi()) {
            return $event->getConnectionCode()
        }

        $connection = $this->connectionContext->getConnection();

        return null !== $connection ? (string) $connection->code() : null;
```

Reply:
Also, you only return null to throw right after if null ?
Why not throw directly here if there is no connection code ?
Your method could then return only `string`, not `?string`.